### ISSUE TITLE
How can I make Yacy use https using Namecheap?

### ISSUE THREAD

**Issue by songproducer** [yacy/yacy_search_server#194]
They're asking for a CSR
https://www.namecheap.com/support/knowledgebase/article.aspx/467/67/how-do-i-generate-a-csr-code

I'm not really sure following this guide:
http://www.yacy-websuche.de/wiki/index.php/En:HOWTO_make_YaCy_allow_SSL_connections

**Comment by luccioman** [comment]
Hi @songproducer , personally I never dealt with Namecheap, but the instructions look reasonable. YaCy wiki page is also ok. You also have short instructions in the [yacy.init](https://github.com/yacy/yacy_search_server/blob/f4267ed247361916eb937ca7121d1d2f8e101015/defaults/yacy.init#L36) file itself.
Please note also that for example [Let's Encrypt](https://letsencrypt.org/) may be an interesting alternative to get a free an valid certificate.
And also [KeyStore Explorer](http://keystore-explorer.org/) is to my mind a good graphical tool to easily deal with your keys/certificates.

**Comment by songproducer** [comment]
I'm not having much luck working out how to set up Let's Encrypt with Yacy.

So far I've done this:
* In defaults/yacy.conf, set:
server.https=true
* Installed certbot
* Ran sudo certbot --apache (which fails because yacy doesn't seem to be on https yet)
* Installed KeyStore Explorer (not sure how to use)
* Forwarded port 443 on router to my computer IP
* Ran:
keytool -keystore mySrvKeystore -genkey -keyalg RSA -alias mycert
* set up password and stored in defaults/mySrvKeystore
* updated yacy.conf:
keyStore=defaults/mySrvKeystore
keyStorePassword=password

As you can see I'm in way over my head. I once got Yacy running in https about a year ago but I've forgotten how to do it.

**Comment by songproducer** [comment]
@Scarfmonster Thanks! However, when I run:
sudo certbot certonly --standalone -d bitcoind.party
I get:
Problem binding to port 80: Could not bind to IPv4 or IPv6.

**Comment by Scarfmonster** [comment]
@songproducer That means there is something running on port 80 already. Ane you running a web server on the machine as well? Or maybe redirecting yacy to run on port 80?

**Comment by songproducer** [comment]
@Scarfmonster Yes, I was redirecting requests from port 80 to 8090. I just worked out that was why certbot was failing so now I've got the lets encrypt cert working. But https is still not working.

Do I just need to fill in pkcs12ImportFile and pkcs12ImportPwd or do I need to do keyStore and keyStorePassword too?

**Comment by songproducer** [comment]
When I go https://localhost/Status.html I get The requested URL /index.html was not found on this server.

**Comment by Scarfmonster** [comment]
@songproducer That's not a response from YaCy, I think. If SSL was configured wrong in YaCy you would get an error about the certificate or connection refused, not an information about a file not found. Try opening https://localhost:8443/Status.html , if it opens then SSL works fine in YaCy.

**Comment by songproducer** [comment]
@Scarfmonster Unfortunately I get Unable to connect

Firefox can’t establish a connection to the server at localhost:8443

**Comment by songproducer** [comment]
An error occurred during a connection to localhost:8433. SSL received a record that exceeded the maximum permissible length. Error code: SSL_ERROR_RX_RECORD_TOO_LONG

**Comment by r4dh4l** [comment]
I'm syncing my LE certificates from the hypervisor to all running VMs to avoid installing certbot on each VM. Is there a way to tell YaCy just to use the imported LE certificates, so no to use a pkcs12 file?

**Comment by r4dh4l** [comment]
> I'm syncing my LE certificates from the hypervisor to all running VMs to avoid installing certbot on each VM. Is there a way to tell YaCy just to use the imported LE certificates, so no to use a pkcs12 file but the `privkey.pem` and `fullchain.pem` file under `/etc/letsencrypt/archive/my.ddns.net/`?

https://github.com/yacy/yacy_search_server/issues/124#issuecomment-437803675

**Comment by o92** [comment]
port=<any local port you wish>
......
port.ssl=443 
....
server.https = true


Apart from that, I followed the same steps as  @r4dh4l mentioned.  No issues using LetsEncrypt certs for my self hosted instance.  Thanks to all the people above for your tips. 

I referred to defaults/yacy.init file for anything that I wasn't clear about.  That file contains some very helpful comments.
